src/renderer/components/swap/Swap.tsx

```diff
@@ -390,6 +389,156 @@ export const Swap = ({
+        {intl.formatMessage({ id: 'swap.errors.amount.balanceShouldCoverChainFee' }, { asset: sourceChainAsset })}{' '}
+      </Styled.BalanceErrorLabel>
+    )
+  }, [sourceChainBalanceError, sourceAsset, intl])
+
+  const targetChainFeeAmountInTargetAsset: BaseAmount = useMemo(() => {
+    const chainFee = FP.pipe(
+      targetChainFee,
+      RD.getOrElse(() => ZERO_BASE_AMOUNT)
+    )
+
+    return FP.pipe(
+      targetAsset,
+      O.map((asset) => {
+        const chainAsset = getChainAsset(asset.chain)
+        return getValueOfAsset1InAsset2(chainFee, poolData[assetToString(chainAsset)], poolData[assetToString(asset)])
+      }),
+      O.getOrElse(() => ZERO_BASE_AMOUNT)
+    )
+  }, [targetChainFee, targetAsset, poolData])
+
+  const targetChainFeeError = useMemo((): boolean => {
+    if (changeAmount.isZero()) {
+      return false
+    }
+    const targetFee = FP.pipe(targetChainFeeAmountInTargetAsset, baseToAsset, (assetAmount) => assetAmount.amount())
+    return swapData.swapResult.minus(targetFee).isNegative()
+  }, [targetChainFeeAmountInTargetAsset, swapData, changeAmount])
+
+  const targetChainFeeErrorLabel = useMemo(() => {
+    if (!targetChainFeeError) {
+      return null
+    }
+
+    const feeAssetAmount = baseToAsset(targetChainFeeAmountInTargetAsset)
+
+    const feeLabel = FP.pipe(
+      targetAsset,
+      O.map((asset) => formatAssetAmountCurrency({ amount: feeAssetAmount, asset, trimZeros: true })),
+      O.getOrElse(() => formatBN(feeAssetAmount.amount()))
+    )
+
+    return (
+      <Styled.ErrorLabel>
+        {intl.formatMessage({ id: 'swap.errors.amount.outputShouldCoverChainFee' }, { fee: feeLabel })}
+      </Styled.ErrorLabel>
+    )
+  }, [targetChainFeeError, targetChainFeeAmountInTargetAsset, intl, targetAsset])
+
+  const fees: RD.RemoteData<Error, Fee[]> = useMemo(
+    () =>
+      FP.pipe(
+        sequenceTRD(
+          sourceChainFee,
+          RD.success(targetChainFeeAmountInTargetAsset),
+          RD.fromOption(sourceAsset, () => Error('No source asset')),
+          RD.fromOption(targetAsset, () => Error('No target asset'))
+        ),
+        RD.map(([sourceFee, targetFee, sourceAsset, targetAsset]) => [
+          { asset: getChainAsset(sourceAsset.chain), amount: sourceFee },
+          { asset: targetAsset, amount: targetFee }
+        ])
+      ),
+    [sourceChainFee, targetChainFeeAmountInTargetAsset, sourceAsset, targetAsset]
+  )
+
+  const isSwapDisabled: boolean = useMemo(
+    () => changeAmount.eq(0) || FP.pipe(walletBalances, O.isNone) || sourceChainBalanceError || targetChainFeeError,
+    [walletBalances, changeAmount, sourceChainBalanceError, targetChainFeeError]
+  )
+
+  const outputLabel = useMemo(
+    () =>
+      FP.pipe(
+        targetAsset,
+        O.map((asset) =>
+          formatAssetAmountCurrency({ amount: assetAmount(swapData.swapResult), asset, trimZeros: true })
+        ),
+        O.getOrElse(() => formatBN(swapData.swapResult, 7))
```

Comment:
Why is decimal `7` ?

Reply:
this was copy-pasted from prev implementation
https://github.com/thorchain/asgardex-electron/pull/663/files#diff-35dddf5143f357d65494359d2b087faeca6e8eb13706b5a652f73f5dfe3a86eeL440

Reply:
If we don't have an asset (it's still needed to swap anything), then we should just return ZERO_BN I think ...

Reply:
i suppose `swapData.swapResult` should contain this data. in case it calculated anything we should show it i believe

Reply:
Please just fix it (as [you suggested](https://github.com/thorchain/asgardex-electron/pull/663#discussion_r542635609) or [me before](https://github.com/thorchain/asgardex-electron/pull/663#discussion_r542619425))

Reply:
@veado changed by [this commit](https://github.com/thorchain/asgardex-electron/pull/663/commits/68b7df576a586d70289e73f68944b115bdcb0caa)